File: WebAPP/ClientApp/src/dto/stayActivity.ts
import { Activity } from "./activity";
import { Stay } from "./stay";

export class StayActivity {
    public id: number;
    public stayId: number;
    public stay: Stay;
    public activityId: number;
    public activity: Activity

    constructor(id?: number, stayId?: number, stay?: Stay, activityId?: number, activity?: Activity)

    constructor(id: number, stayId: number, stay: Stay, activityId: number, activity: Activity) {
        this.id = id;
        this.stayId = stayId;
        this.stay = stay;
        this.activityId = activityId;
        this.activity = activity;
    }
}

File: Model/StayActivity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Model
{
    [Table("stay_activity")]
    public class StayActivity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public int? Id { set; get; }

        [Column("stay_id")]
        public int? StayId { set; get; }
        public Stay Stay { set; get; }

        [Column("activity_id")]
        public int? ActivityId { set; get; }
        public Activity Activity { set; get; }
    }
}
